test/images/agnhost/README.md

```diff
@@ -72,8 +62,213 @@ created with the `pause` argument instead, allowing us execute multiple commands
     kubectl exec test-agnhost -- /agnhost dns-server-list
 ```
 
+The `agnhost` binary is a CLI with the following subcommands:
+
+
+### dns-server-list
+
+It will output the host's configured DNS servers, separated by commas.
+
+Usage:
+
+```console
+    kubectl exec test-agnhost -- /agnhost dns-server-list
+```
+
+
+### dns-suffix
+
+It will output the host's configured DNS suffix list, separated by commas.
+
+Usage:
+
+```console
+    kubectl exec test-agnhost -- /agnhost dns-suffix
+```
+
+
+### etc-hosts
+
+It will output the contents of host's `hosts` file. This file's location is `/etc/hosts`
+on Linux, while on Windows it is `C:/Windows/System32/drivers/etc/hosts`.
+
+Usage:
+
+```console
+    kubectl exec test-agnhost -- /agnhost etc-hosts
+```
+
+
+### fake-gitserver
+
+Fakes a git server. When doing `git clone localhost:8000`, you will clone an empty git
+repo named `8000` on local. You can also use `git clone localhost:8000 my-repo-name` to
+rename that repo.
+
+Usage:
+
+```console
+    kubectl exec test-agnhost -- /agnhost fake-gitserver
+```
+
+
+### help
+
+Prints the binary's help menu. Additionally, it can be followed by another subcommand
+in order to get more information about that subcommand, including its possible arguments.
+
+Usage:
+
+```console
+    kubectl exec test-agnhost -- /agnhost help
+```
+
+
+### liveness
+
+Starts a simple server that is alive for 10 seconds, then reports unhealthy for the rest
+of its (hopefully) short existence.
+
+Usage:
+
+```console
+    kubectl exec test-agnhost -- /agnhost liveness
+```
+
+
+### logs-generator
+
+The `logs-generator` subcommand is a tool to create predictable load on the logs delivery system.
+It generates random lines with predictable format and predictable average length.
+Each line can be later uniquely identified to ensure logs delivery.
+
+Tool is parametrized with the total number of number that should be generated and the duration of
+the generation process. For example, if you want to create a throughput of 100 lines per second
+for a minute, you set total number of lines to 6000 and duration to 1 minute.
+
+Parameters are passed through environment variables. There are no defaults, you should always
+set up container parameters. Total number of line is parametrized through env variable
+`LOGS_GENERATOR_LINES_TOTAL` and duration in go format is parametrized through env variable
+`LOGS_GENERATOR_DURATION`.
```

Comment:
this paragraph is incorrect: near as I can tell, `logs_generator.go` only takes flags, not env vars, and I can't tell that agnhost plumbs the env vars through?

Reply:
This part of the documentation was pretty much copied from the original documentation logs-generator had. Interestingly, it was like this from the beggining: accepting those parameters as flags and not as env vars, yet the documentation is talking about env vars. The tests also sets those env vars. In the replacement PR, I've replaced that with actual flags: https://github.com/kubernetes/kubernetes/pull/78394/files#diff-7dc1ebd3b2f2f6ca1ff248e6601016a6R108